includes/wikia/parser/templatetypes/TemplateTypesParser.class.php

```diff
@@ -35,20 +34,28 @@ public static function onFetchTemplateAndTitle( &$text, &$finalTitle ) {
 	}
 
 	/**
-	 * @desc change template wikitext according to template type
+	 * @desc alters template parser output based on its arguments and template type
+	 *
+	 * @param Title $title
+	 * @param PPNode_DOM $args
+	 * @param PPFrame_DOM $frame
+	 * @param string $outputText
 	 *
-	 * @param $templateTitle
-	 * @param $templateWikitext
 	 * @return bool
 	 */
-	public static function onBraceSubstitution( $templateTitle, &$templateWikitext ) {
+	public static function onGetTemplateDom( $title, $args, $frame, &$outputText ) {
+		global $wgEnableTemplateTypesParsing, $wgEnableScrollboxTemplateParsing, $wgArticleAsJson;
 		wfProfileIn( __METHOD__ );
-		$title = Title::newFromText( $templateTitle, NS_TEMPLATE );
 
-		if ( self::templateShouldBeProcessed( $templateWikitext ) && self::isValidTitle( $title ) ) {
+		if ( $wgEnableTemplateTypesParsing && $wgEnableScrollboxTemplateParsing && $wgArticleAsJson && !isNull($args) &&
```

Comment:
I am in general greatly opposed to compound "if" statements. Can this be shortened to a single if statement as before to make sure the logic behind it is in one place?